htdocs/categories/class/categorie.class.php

```diff
@@ -1894,14 +1921,16 @@ public static function getFilterSelectQuery($type, $rowIdName, $searchList)
 		$searchCategorySqlList = array();
 		foreach ($searchList as $searchCategory) {
 			if (intval($searchCategory) == -2) {
-				$searchCategorySqlList[] = " cp.fk_categorie IS NULL";
+				$searchCategorySqlList[] = " $categAlias.fk_categorie IS NULL";
 			} elseif (intval($searchCategory) > 0) {
-				$searchCategorySqlList[] = " ".$rowIdName." IN (SELECT fk_".$type." FROM ".MAIN_DB_PREFIX."categorie_".$type." WHERE fk_categorie = ".((int) $searchCategory).")";
+				$searchCategorySqlList[] = " ".$rowIdName." IN (SELECT fk_".$fk." FROM ".MAIN_DB_PREFIX."categorie_".$tableSuffix." WHERE fk_categorie = ".$searchCategory.")";
```

Comment:
Why removing the (int) on searchCategory (the cat is here to be sure we do not inject bad sql string from a forged searchCategory string.